great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -214,59 +225,93 @@ def _init_domain_builder(
+        expectation_configuration_builder_configs: List[dict],
     ) -> List[ExpectationConfigurationBuilder]:
         expectation_configuration_builders: List[ExpectationConfigurationBuilder] = []
+
+        expectation_configuration_builder_config: dict
         for (
             expectation_configuration_builder_config
         ) in expectation_configuration_builder_configs:
-            expectation_configuration_builder = instantiate_class_from_config(
-                config=expectation_configuration_builder_config,
-                runtime_environment={},
-                config_defaults={
-                    "class_name": "DefaultExpectationConfigurationBuilder",
-                    "module_name": "great_expectations.rule_based_profiler.expectation_configuration_builder",
-                },
+            expectation_configuration_builder: ExpectationConfigurationBuilder = RuleBasedProfiler._init_one_expectation_configuration_builder(
+                expectation_configuration_builder_config=expectation_configuration_builder_config,
             )
             expectation_configuration_builders.append(expectation_configuration_builder)
 
         return expectation_configuration_builders
 
+    @staticmethod
+    def _init_one_expectation_configuration_builder(
+        expectation_configuration_builder_config: dict,
+    ) -> ExpectationConfigurationBuilder:
+        expectation_configuration_builder: ExpectationConfigurationBuilder = instantiate_class_from_config(
+            config=expectation_configuration_builder_config,
+            runtime_environment={},
+            config_defaults={
+                "class_name": "DefaultExpectationConfigurationBuilder",
+                "module_name": "great_expectations.rule_based_profiler.expectation_configuration_builder",
+            },
+        )
+        return expectation_configuration_builder
+
     def run(
         self,
-        *,
+        variables: Optional[Dict[str, Any]] = None,
+        rules: Optional[Dict[str, Dict[str, Any]]] = None,
         expectation_suite_name: Optional[str] = None,
         include_citation: bool = True,
     ) -> ExpectationSuite:
         """
         Args:
+            :param variables attribute name/value pairs (overrides)
+            :param rules name/(configuration-dictionary) (overrides)
             :param expectation_suite_name: A name for returned Expectation suite.
             :param include_citation: Whether or not to include the Profiler config in the metadata for the ExpectationSuite produced by the Profiler
         :return: Set of rule evaluation results in the form of an ExpectationSuite
         """
+        effective_variables: Optional[ParameterContainer] = self._reconcile_variables(
+            variables=variables
+        )
+
+        # TODO: <Alex>ALEX -- Tests for Reconciliation are next immediate action items.</Alex>
+        effective_rules: List[Rule] = self.reconcile_rules_for_profiler(rules=rules)
```

Comment:
Will this be part of this PR or a later one?

Reply:
@cdkini I would prefer to unblock the rest for now and merge; my very next is these tests (per standup decision last Thursday).  Thanks.

Reply:
okay I agree with that - thank you!